crates/core/src/network/reducers/init.rs

```diff
@@ -156,4 +158,38 @@ pub mod test {
 
         assert_eq!(result, ());
     }
+
+    #[test]
+    pub fn should_set_sim2h_url() {
+        let p2p_config = P2pConfig::new_with_sim2h_backend("wss://localhost:9999");
+        let context: Arc<Context> = test_context(p2p_config);
+        let dna_address: Address = context.agent_id.address();
+        let agent_id = context.agent_id.content().to_string();
+        let handler = NetHandler::new(Box::new(|_| Ok(())));
+        let network_settings = crate::action::NetworkSettings {
+            p2p_config: context.p2p_config.clone(),
+            dna_address,
+            agent_id,
+            handler,
+        };
+        let action_wrapper = ActionWrapper::new(Action::InitNetwork(network_settings));
+
+        let mut network_state = NetworkState::new();
+        let mut root_state = test_store(context.clone());
+
+        let props = json!({"sim2h_url": "wss://localhost:9000"});
+        let mut dna = Dna::new();
+        dna.properties = props;
+
+        root_state = root_state.reduce(ActionWrapper::new(Action::InitializeChain(dna)));
+
+        let result = reduce_init(&mut network_state, &root_state, &action_wrapper);
+
+        assert_eq!(result, ());
+
+        if let Some(network) = network_state.network {
+            assert_eq!(network.p2p_endpoint().as_str(), "wss://localhost:9000/")
+        } else {
+        }
```

Comment:
I think this can be simplified to the following since we actually expect `network` to be `Some`:
```rust
let network = network_state.network.unwrap();
assert_eq!(network.p2p_endpoint().as_str(), "wss://localhost:9000/");
```
Also, I would suggest to make the changed URL differ more - I almost overlooked that the port is different.